components/eamxx/src/physics/ml_correction/eamxx_ml_correction_process_interface.cpp

```diff
@@ -86,16 +91,23 @@ void MLCorrection::run_impl(const double dt) {
   // use model time to infer solar zenith angle for the ML prediction
   auto current_ts = timestamp();
   std::string datetime_str = current_ts.get_date_string() + " " + current_ts.get_time_string();
+
+  const auto &phis            = get_field_in("phis").get_view<const Real *, Host>();
+  const auto &sfc_alb_dif_vis = get_field_in("sfc_alb_dif_vis").get_view<const Real *, Host>();  
+
   const auto &qv              = get_field_out("qv").get_view<Real **, Host>();
   const auto &T_mid           = get_field_out("T_mid").get_view<Real **, Host>();
-  const auto &phis            = get_field_in("phis").get_view<const Real *, Host>();
   const auto &SW_flux_dn      = get_field_out("SW_flux_dn").get_view<Real **, Host>();
-  const auto &sfc_alb_dif_vis = get_field_in("sfc_alb_dif_vis").get_view<const Real *, Host>();  
   const auto &sfc_flux_sw_net = get_field_out("sfc_flux_sw_net").get_view<Real *, Host>();
   const auto &sfc_flux_lw_dn  = get_field_out("sfc_flux_lw_dn").get_view<Real *, Host>();
   const auto &u               = get_field_out("horiz_winds").get_component(0).get_view<Real **, Host>();
   const auto &v               = get_field_out("horiz_winds").get_component(1).get_view<Real **, Host>();
 
+  // For precipitation adjustment we need to track the change in column integrated 'qv'
+  auto qv_told = Kokkos::create_mirror_view(qv);
```

Comment:
This won't work. `qv_told` will just be an alias for `qv`, since they are both on host. What you need is something like
```c++
host_view2d_type qv_told("",qv.extent(0),qv.extent(1));
Kokkos::deep_copy(qv_told,qv);
```

Reply:
thanks for the fix.